lib/core/versions/latest/DocumentComposer.ts

```diff
@@ -40,7 +40,7 @@ export default class DocumentComposer {
         const id = '#' + publicKey.id;
         const didDocumentPublicKey = {
           id: id,
-          controller: '',
+          controller: did.shortForm,
```

Comment:
For accuracy we can't have this always be hardcoded to short form (in case of a long form DID). I'd imagine you'll have to do something like:

```
did.isShortForm ? did.shortForm : did.longForm;
```

Reply:
I was thinking about switching to this, but this would imply that the long form identifier is different than the short form identifier based on which state is being resolved. This seemed off to me, so I opted to keep it always set to the shortForm which as I understood it was always available since it's just a shorter instance of the long form. What's the advantage you see to having the long form identifier available here?

Reply:
Long form DIDs will indicate that the short form is an equivilant DID in its metadata

```   
 "didDocumentMetadata": {
      "deactivated": true,
      "canonicalId": "did:sidetree:EiDyOQbbZAa3aiRzeCkV7LOx3SERjjH93EXoIM3UoN4oWg",
      "equivalentId": ["did:sidetree:EiDyOQbbZAa3aiRzeCkV7LOx3SERjjH93EXoIM3UoN4oWg"],
      "method": {
        "published": true,
        "recoveryCommitment": "EiBfOZdMtU6OBw8Pk879QtZ-2J-9FbbjSZyoaA_bqD4zhA",
        "updateCommitment": "EiDOrcmPtfMHuwIWN6YoihdeIPxOKDHy3D6sdMXu_7CN0w"
      }
    }
```

I think including the long form ID here when the resolving DID is long form is better, since someone could try to resolve the controller's DID and the short form can potentially be not found. The resolving party can always look at the meta data of the long form DID document and decide to resolve long form or short form so I don't think we will have circular dependency. I imagine something like

1. resolve a long form DID
2. controller is long form
3. try to resolve short form
4.a. short form exists, use short form to continue and use new controller which is short form
4.b. short form does not exist, use long form and the controller is still valid